### ISSUE TITLE
Codegen generated with 1.1.0-beta1 is showing compilation errors in Visual Studio 2015

### ISSUE THREAD

**Issue by shanegrueling** [dotnet/orleans#1033]
Hey,

when using the new pre-release version the codegen generates wrong code for structs.

Example struct:
```
public struct GameMove
    {
        public Guid PlayerId { get; set; }
        public int X { get; set; }
        public int Y { get; set; }
    }
```

Error shown in Visual Studio 2015 is:
```
Error    CS1061    'object' does not contain a definition for 'PlayerId' and no extension method 'PlayerId' accepting a first argument of type 'object' could be found (are you missing a using directive or an assembly reference?)
Error    CS1061    'object' does not contain a definition for 'X' and no extension method 'X' accepting a first argument of type 'object' could be found (are you missing a using directive or an assembly reference?)
Error    CS1061    'object' does not contain a definition for 'Y' and no extension method 'Y' accepting a first argument of type 'object' could be found (are you missing a using directive or an assembly reference?)
```

and generated Code is:

```
 public static global::System.Object DeepCopier(global::System.Object original)
        {
            global::GrainInterfaces.GameMove input = ((global::GrainInterfaces.GameMove)original);
            global::GrainInterfaces.GameMove result = default (global::GrainInterfaces.GameMove);
            global::System.Object boxedResult = result;
            global::Orleans.@Serialization.@SerializationContext.@Current.@RecordObject(original, boxedResult);
            boxedResult.@PlayerId = input.@PlayerId;
            boxedResult.@X = input.@X;
            boxedResult.@Y = input.@Y;
            return boxedResult;
        }
```

**Comment by ReubenBond** [comment]
Thank you for reporting, @shanegrueling!

This is resolved by #1028 :)

**Comment by gabikliot** [comment]
Fixed via #1028.
